ValveResourceFormat/IO/MaterialExtract.cs

```diff
@@ -0,0 +1,482 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using ValveKeyValue;
+using ValveResourceFormat.Blocks;
+using ValveResourceFormat.ResourceTypes;
+
+namespace ValveResourceFormat.IO;
+
+using ChannelMapping = ValueTuple<Channel, string>;
+
+public struct UnpackInfo
+{
+    public string TextureType { get; init; }
+    public string FileName { get; init; }
+    public Channel Channel { get; init; }
+}
+
+public enum Channel
```

Comment:
`ColorChannel` or something? Channel is a bit generic in IO namespace. Or put these two inside of MaterialExtract.

Reply:
Putted them both inside MaterialExtract. I need Channel to be short